### PR TITLE
6213 Барышников В. С. Лаб. 5 Вар. 1

### PR DESCRIPTION
**Pull request by dedrowe**
В данной лабораторной требовалось обучить и протестировать модель нейронной сети.
Ссылка на коллаб: https://colab.research.google.com/drive/1Yfi-vLO8-aNQ98UBthSg-z__RHA_jUSA?usp=sharing

### CODE REVIEW

File: Lab5/Лаб_5_Вар_1_Барышников_6213.ipynb [hunk 1]
```diff
@@ -0,0 +1,1595 @@
...
+        "\n",
+        "loss_fn = nn.MSELoss(reduction=\"mean\")\n",
+        "optimizer = optim.Adam(model.parameters(), lr=learning_rate, weight_decay=weight_decay)\n",
+        "\n",
+        "\n",
+        "opt = Optimization(model=model, loss_fn=loss_fn, optimizer=optimizer)\n",
+        "opt.train_loop(train_loader, val_loader, batch_size=batch_size, n_epochs=n_epochs, n_features=input_dim)\n",
+        "opt.plot_losses()\n",
+        "\n",
+        "predictions, values = opt.evaluate(\n",
```
Comment (alxmcs) [inline Lab5/Лаб_5_Вар_1_Барышников_6213.ipynb:1154]: То, что validation loss = 0 совершенно никак тебя не напрягло?

**Review by alxmcs — CHANGES_REQUESTED** [review]
Судя по всему, ты и #60 смотрели в один и тот же туториал и допустили один и тот же косяк

**Review by alxmcs — CHANGES_REQUESTED** [review]
По предсказанным значениям видно, что сеть стала работать еще хуже.
image

Очевидно, это происходит, что при распределении данных для обучения и валидации - у тебя стоит `shuffle=False`, в итоге ты учишь сеть данными старых лет, а для валидации подсовываешь современные.

В итоге на графике функции потерь у тебя сначала куча скачков у обеих кривых,а потом оверфит. Невооруженным взглядом видно, что это не здоровая кривая:
image

Поэтому предлагаю, во первых, `shuffle=True`, во вторых - исключить из рассмотрения данные до деноминации.

**Review by alxmcs — APPROVED** [review]
